cadquery/assembly.py

```diff
@@ -231,24 +258,40 @@ def _query(self, q: str) -> Tuple[str, Optional[Shape]]:
         Execute a selector query on the assembly. 
         The query is expected to be in the following format:
         
-            name@kind@args
+            name[?tag][@kind@args]
             
-        for example:
+        valid example include:
         
-            obj_name@faces@>Z
+            obj_name @ faces @ >Z
+            obj_name?tag1@faces@>Z          
+            obj_name ? tag
+            obj_name
         
         """
 
-        name, kind, arg = q.split("@")
+        tmp: Workplane
+        res: Workplane
+
+        query = _grammar.parseString(q, True)
+        name: str = query.name
 
-        tmp = Workplane()
         obj = self.objects[name].obj
 
-        if isinstance(obj, (Workplane, Shape)):
-            tmp.add(obj)
-            res = getattr(tmp, kind)(arg)
+        if isinstance(obj, Workplane) and query.tag:
+            tmp = obj.ctx.tags[query.tag]
```

Comment:
I'm a bit late here, but wouldn't it be better to use `tmp = obj._getTagged(query.tag)`? Try to keep a single method to access tags to be DRY, consistent behaviour (note this will raise a KeyError, while the rest of CQ will raise a ValueError with a bad tag), etc.

Otherwise looks great to me.

Reply:
Good point, will fix it together with #469 .